pkg/cloudprovider/providers/azure/azure_wrap.go

```diff
@@ -54,25 +56,72 @@ func ignoreStatusNotFoundFromError(err error) error {
 	return err
 }
 
+// cache used by getVirtualMachine
+var vmCache = cache.New(15*time.Second, 10*time.Minute)
+var vmCacheLock = &sync.Mutex{}
```

Comment:
Remove this lock? go-cache has already using a lock inside each Get/Set interface